src/newmatrix.cpp

```diff
@@ -1864,36 +1880,41 @@ ACMD(do_connect)
   if (!(jack = get_datajack(ch, FALSE)))
     return;
 
-  for (temp = ch->in_room->people; temp; temp = temp->next_in_room)
-    if (PLR_FLAGGED(temp, PLR_MATRIX) && !IS_IGNORING(temp, is_blocking_ic_interaction_from, ch)) {
-      if (temp->persona && temp->persona->decker->deck) {
-        for (struct obj_data *hitch = temp->persona->decker->deck->contains; hitch; hitch = hitch->next_content) {
-          if (GET_OBJ_TYPE(hitch) == ITEM_DECK_ACCESSORY && GET_OBJ_VAL(hitch, 0) == 1 &&
-              GET_OBJ_VAL(hitch, 1) == 3)
-          {
-            for (struct char_data *temp2 = ch->in_room->people; temp2; temp2 = temp2->next_in_room) {
-              if (temp2 != temp && PLR_FLAGGED(temp2, PLR_MATRIX)) {
-                act("The hitcher jack on $n's deck is already in use.", FALSE, temp, 0, ch, TO_VICT);
-                return;
-              }
-            }
-#ifdef ALLOW_HITCHING
-            act("You slip your jack into $n's hitcher port.", FALSE, temp, 0, ch, TO_VICT);
-            send_to_char("Someone has connected to your hitcher port.\r\n", temp);
-            PLR_FLAGS(ch).SetBit(PLR_MATRIX);
-            temp->persona->decker->hitcher = ch;
-#else
-            send_to_char(ch, "Sorry, the hitching system is disabled right now.\r\n");
-#endif
-            return;
-          }
-        }
-      }
+  // hitcher code, new syntax is connect <dude>
+  if (*argument && !PLR_FLAGGED(ch, PLR_MATRIX)) {
+    skip_spaces(&argument);
+    
+    temp = get_char_room_vis(ch, argument);
+    if (IS_IGNORING(temp, is_blocking_ic_interaction_from, ch)) temp = NULL;
+    if (!temp) {
+      send_to_char(ch, "You don't see anyone named '%s' here.\r\n", argument);
```

Comment:
The goal is for the ignored person to not know they're being ignored, so when the ignore bit is set, please use an error like "It doesn't look like you can hitch a ride with %s" below. Otherwise they'll know that they're being blocked, because the person they're trying to connect to is standing in front of them.

Reply:
Will fix!